README.md

```diff
@@ -48,161 +146,276 @@ To run the demo code, follow these steps:
+
+```js
+const myConn = mySqlEasier.createConnection({
+  host: 'localhost',
+  user: 'myuser',
+  password: 'super-secret'
+});
+```
+
+#### `createPool`
+Creates a new MySqlPool using the provided configuration.
+
+This method can be used as an alternative to `configure`.
+It may be useful if you would like to manage your connection pools yourself.
+For example, if you require multiple simultaneous pools to exist at the same time.
 
-## `deleteAll`
-This deletes all records from a given table.
+When you are done with the pool, you should call `myPool.end()`.
+
+```js
+const myPool = mySqlEasier.createPool({
+  host: 'localhost',
+  user: 'myuser',
+  password: 'super-secret'
+});
+```
+
+#### `endPool`
+Ends the global pool if it was created using `configure`.
+
+Once the pool has ended, any outstanding connections will cease to perform.
+This should only be done when you are confident that the pool is no longer necessary.
+
+After you end the global pool, you can create a new one by calling `mySqlEasier.configure(config)` again.
+
+```js
+try {
+  await mySqlEasier.endPool();
+} catch (err) {
+  // Handle the error.
+}
+```
+
+#### `getConnection`
+Retrieves a MySqlConnection from the global pool.
+
+Once you are done with the connection, please call `myConnection.done()` to release it back to the pool.
 
 ```js
 try {
-  const mysql.deleteAll('flavors');
+  const myConn = await mySqlEasier.getConnection();
+} catch (err) {
+  // Handle the error.
+}
+```
+
+### MySqlConnection
+
+This class wraps a 'Connection' from mysqljs as described here:
+  + https://github.com/mysqljs/mysql#establishing-connections
+
+The basic benefits it provides are:
+  1. `query` and other database methods return promises instead of taking callbacks.
+  2. Provides a number of convenience methods for interacting with tables that have an auto-incremented primary key named `id`.
+
+If you need to do some more advanced operations (like [streaming result rows](https://github.com/mysqljs/mysql#streaming-query-rows)), you can access the [mysqljs](https://github.com/mysqljs/mysql) 'Connection' by going directly to the `connection` property of this object.
+
+Generally speaking, you will create a MySqlConnection directly (but you can if you like).
+You will usually get connections by:
+
+  + `mySqlEasier.getConnection()` -- To grab a connection from the global pool.
+  + `myPool.getConnection()` -- Where `myPool` is a MySqlPool from mysql-easier.
+
+When you are done with the connection, please call `myConnection.done()`.
+If the connection is pooled, it will release the connection back to its pool.
+If it is not, it will just 'end' the connection.
+
+To create a MySqlConnection from an existing mysqljs Connection:
+
+```js
+const easierConnection = new MySqlConnection(baseConnection);
+```
+
+#### `deleteAll`
+Deletes all records from a given table.
+
+```js
+try {
+  const await connection.deleteAll('flavors');
```

Comment:
I don't think you want the "const" here.

Reply:
👍